File: Data Structures & Algorithms/daily-temperatures/submission-16.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> results(temperatures.size(), 0);
        stack<pair<int, int>> storage;

        for (int i = 0; i < temperatures.size(); i++) {
            int current = temperatures[i];

            while (!storage.empty() && storage.top().second < current) {
                int index = storage.top().first;
                results[index] = i - index;
                storage.pop();
            }

            storage.push({i, current});
        }

        return results;
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-16.py
class Solution:
    def isValid(self, s: str) -> bool:
        closeToOpen = {')': '(', '}': '{', ']': '['}
        stack = []

        for bracket in s:
            if bracket in closeToOpen:
                if stack and stack[-1] == closeToOpen[bracket]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(bracket)
        
        return not stack